app/http/httpclient.c

```diff
@@ -451,20 +451,33 @@ static void ICACHE_FLASH_ATTR http_timeout_callback( void *arg )
 	struct espconn * conn = (struct espconn *) arg;
 	if ( conn == NULL )
 	{
+		HTTPCLIENT_DEBUG( "Connection is NULL" );
 		return;
 	}
 	if ( conn->reverse == NULL )
 	{
+		HTTPCLIENT_DEBUG( "reverse is NULL" );
```

Comment:
@marcelstoer As to the documentation inconsistency: I am not aware of it, please enlighten me so I can add it.

The fix above establishes the behavior as in the docs AFAICS.